From 42df451ec7b95691f07d03ee8a3e4fb0d6e1acb2 Mon Sep 17 00:00:00 2001
From: Guannan Ren <gren@redhat.com>
Date: Thu, 5 Sep 2013 02:49:41 +0800
Subject: [PATCH 2/8] error: use helper function to embed customized widget
 into dialog

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=807277
(cherry picked from commit d6fc079ae88da2433ae257b25e58f48403b78cf7)

Conflicts:
	src/virtManager/error.py #fix trival conflicts like Gtk vs gtk
---
 src/virtManager/error.py | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/virtManager/error.py b/src/virtManager/error.py
index 28d8859..7870061 100644
--- a/src/virtManager/error.py
+++ b/src/virtManager/error.py
@@ -31,11 +31,14 @@ def safe_set_text(self, text):
         self.set_markup(text)
 
 def _launch_dialog(dialog, primary_text, secondary_text, title,
-                   sync=True):
+                   widget=None, sync=True):
     safe_set_text(dialog, primary_text)
     dialog.format_secondary_text(secondary_text or None)
     dialog.set_title(title)
 
+    if widget:
+        dialog.get_content_area().add(widget)
+
     res = False
     if sync:
         res = dialog.run()
@@ -92,7 +95,7 @@ class vmmErrorDialog(vmmGObject):
     ###################################
 
     def _simple_dialog(self, dialog_type, buttons, text1,
-                       text2, title, async=False):
+                       text2, title, widget=None, async=False):
 
         dialog = gtk.MessageDialog(self.get_parent(),
                                    gtk.DIALOG_DESTROY_WITH_PARENT,
@@ -103,6 +106,7 @@ class vmmErrorDialog(vmmGObject):
 
         return _launch_dialog(self._simple,
                               text1, text2 or "", title or "",
+                              widget=widget,
                               sync=not async)
 
     def val_err(self, text1, text2=None, title=_("Input Error"), async=True):
@@ -112,10 +116,10 @@ class vmmErrorDialog(vmmGObject):
         self._simple_dialog(dtype, buttons, text1, text2, title, async)
         return False
 
-    def show_info(self, text1, text2=None, title="", async=True):
+    def show_info(self, text1, text2=None, title="", widget=None, async=True):
         dtype = gtk.MESSAGE_INFO
         buttons = gtk.BUTTONS_OK
-        self._simple_dialog(dtype, buttons, text1, text2, title, async)
+        self._simple_dialog(dtype, buttons, text1, text2, title, widget, async)
         return False
 
     def yes_no(self, text1, text2=None, title=None):
-- 
1.9.0

